File: ClockWorkerCleaning/backend/src/utils/response.js
function success(data = null, message = 'success') {
  return {
    code: 0,
    message,
    data,
  };
}

function error(message = 'error', code = 1, data = null) {
  return {
    code,
    message,
    data,
  };
}

function pagination(list, total, page, pageSize) {
  return {
    list,
    total,
    page: parseInt(page),
    pageSize: parseInt(pageSize),
    totalPages: Math.ceil(total / pageSize),
  };
}

module.exports = {
  success,
  error,
  pagination,
};

File: ElectricBicycleBatteryReplacementCabinet/internal/pkg/response/response.go
package response

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

const (
	CodeSuccess = 0
	CodeError   = 1
)

func Success(c *gin.Context, data interface{}) {
	c.JSON(http.StatusOK, Response{
		Code:    CodeSuccess,
		Message: "success",
		Data:    data,
	})
}

func Error(c *gin.Context, message string) {
	c.JSON(http.StatusOK, Response{
		Code:    CodeError,
		Message: message,
	})
}

func ErrorWithCode(c *gin.Context, code int, message string) {
	c.JSON(http.StatusOK, Response{
		Code:    code,
		Message: message,
	})
}

func PageResult(list interface{}, total int64, page, pageSize int) map[string]interface{} {
	return map[string]interface{}{
		"list":      list,
		"total":     total,
		"page":      page,
		"page_size": pageSize,
		"pages":     (total + int64(pageSize) - 1) / int64(pageSize),
	}
}
